src/utils/frame.js

```diff
@@ -160,6 +170,49 @@ export function annotateFrame(frame: Frame) {
   return frame;
 }
 
+function annotateBabelAsyncFrames(frames: Frame[]) {
+  const babelFrameIndexes = getBabelFrameIndexes(frames);
+  const isBabelFrame = frameIndex => babelFrameIndexes.includes(frameIndex);
+
+  return frames.map(
+    (frame, frameIndex) =>
+      isBabelFrame(frameIndex) ? { ...frame, library: "Babel" } : frame
+  );
+}
+
+function getBabelFrameIndexes(frames) {
+  const startIndexes = getFrameIndices(
+    frames,
+    (displayName, url) =>
+      url.match(/regenerator-runtime/i) && displayName === "tryCatch"
+  );
+
+  const endIndexes = getFrameIndices(
+    frames,
+    (displayName, url) =>
+      displayName === "_asyncToGenerator/<" ||
+      (url.match(/_microtask/i) && displayName === "flush")
+  );
+
+  if (startIndexes.length != endIndexes.length || startIndexes.length === 0) {
+    return frames;
+  }
+
+  return flatMap(zip(startIndexes, endIndexes), ([start, end]) =>
+    range(start, end + 1)
+  );
```

Comment:
hmm, perhaps `start` should be `startIndex`

Reply:
but yeah, it is hard to see how `[[1,3], [5,8]]` might expand to `[[1,2,3], [5,6,7,8]]` but it is also hard to see how it would fail, once you understand what it does.

There are no obvious bugs, and once you study it, it is obviously correct... :wink: